Sources/ZcashLightClientKit/Rust/ZcashRustBackend.swift

```diff
@@ -281,46 +281,46 @@ struct ZcashRustBackend: ZcashRustBackendWelding {
     }
 
     @DBActor
-    func getTransparentBalance(account: Int32) async throws -> Int64 {
-        guard account >= 0 else {
-            throw ZcashError.rustGetTransparentBalanceNegativeAccount(Int(account))
+    func getTransparentBalance(account: Zip32Account) async throws -> Int64 {
+        guard account.index >= 0 else {
+            throw ZcashError.rustGetTransparentBalanceNegativeAccount(Int(account.index))
         }
 
         let balance = zcashlc_get_total_transparent_balance_for_account(
             dbData.0,
             dbData.1,
             networkType.networkId,
-            account
+            account.index
         )
 
         guard balance >= 0 else {
             throw ZcashError.rustGetTransparentBalance(
-                Int(account),
-                lastErrorMessage(fallback: "Error getting Total Transparent balance from account \(account)")
+                Int(account.index),
+                lastErrorMessage(fallback: "Error getting Total Transparent balance from account \(account.index)")
             )
         }
 
         return balance
     }
 
     @DBActor
-    func getVerifiedTransparentBalance(account: Int32) async throws -> Int64 {
-        guard account >= 0 else {
-            throw ZcashError.rustGetVerifiedTransparentBalanceNegativeAccount(Int(account))
+    func getVerifiedTransparentBalance(account: Zip32Account) async throws -> Int64 {
+        guard account.index >= 0 else {
+            throw ZcashError.rustGetVerifiedTransparentBalanceNegativeAccount(Int(account.index))
         }
 
```

Comment:
Also delete all occurrences of `rustGetVerifiedTransparentBalanceNegativeAccount`.

```suggestion
    func getVerifiedTransparentBalance(accountIndex: Zip32AccountIndex) async throws -> Int64 {
```

Reply:
Updated